packages/react/src/FieldSet/FieldSet.tsx

```diff
@@ -15,7 +15,7 @@ export type FieldSetProps = HintProps &
     /** Whether the field set has an input with a validation error */
     invalid?: boolean
     /** The text for the caption. */
-    legend: string
+    legend: string | ReactNode
```

Comment:
This is the easiest, I could add something that would require this to be a Heading component.

Reply:
Seeing as the Heading in a `legend` always has to be level 1, shouldn't we just add a prop? Something like `withHeading`?

Also, currently the styling changes when using a heading in a legend, it should not.

Reply:
Just a regular `<h1>` then?